omegaconf/dictconfig.py

```diff
@@ -292,9 +292,7 @@ def _s_validate_and_normalize_key(self, key_type: Any, key: Any) -> DictKeyType:
             return key  # type: ignore
         elif issubclass(key_type, Enum):
             try:
-                ret = EnumNode.validate_and_convert_to_enum(
-                    key_type, key, allow_none=False
-                )
+                ret = EnumNode.validate_and_convert_to_enum(key_type, key)
```

Comment:
@Jasha10, can you double check to see that we are not currently missing a test? Not sure why we can just remove allow_none here without changing relevant tests.

Reply:
This was discussed here: https://github.com/omry/omegaconf/pull/578#discussion_r587994126

Reply:
Yes, this is a clean change because `validate_and_convert_to_enum` function signature has changed too.